zip-0315.rst

```diff
@@ -1,9 +1,629 @@
 ::
 
   ZIP: 315
-  Title: Best Practices for Wallet Handling of Multiple Pools
+  Title: Best Practices for Wallet Implementations
   Owners: Daira Emma Hopwood <daira@electriccoin.co>
           Jack Grigg <jack@electriccoin.co>
-  Status: Reserved
+          Kris Nuttycombe <kris@electriccoin.co>
+  Credits: Francisco Gindre
+  Status: Proposed
   Category: Wallet
   Discussions-To: <https://github.com/zcash/zips/issues/447>
+  Pull-Request: <https://github.com/zcash/zips/issues/>
+
+
+Terminology
+===========
+
+The key words "MUST", "SHOULD", "SHOULD NOT", and "MAY" in this document are to be
+interpreted as described in RFC 2119. [#RFC2119]_
+
+The terms below are to be interpreted as follows:
+
+Auto-shielding
+    The process of automatically transferring transparent funds to the most recent
+    shielded pool on receipt.
+
+Opportunistic shielding
+    The process of transferring previously received transparent funds to the most
+    recent shielded pool as part of a user-initiated transaction.
+
```

Comment:
Removed because we will recommend against opportunistic shielding, and it can be defined at the point that we do that.

```suggestion
```